gen-docs.js

```diff
@@ -1,12 +1,28 @@
 const fs = require("fs");
-fs.readFile(process.argv[process.argv.length - 1], "utf8", (err, data) => {
-    if (err) {
-        return console.log(err);
-    } else {
-        const vars = parseVars(data);
-        printDocs(vars, parseDocs(data));
-    }
+// magic vars
+let stream = fs.createWriteStream("./docs/content/main/magic-vars.md", {
+    flags: "a",
 });
+parseMagicVariables(fs.readFileSync("./src/config/inbuilt.rs", "utf8"), stream);
+stream.write("\n## Static Magic Variables\n\n");
+parseMagicVariables(fs.readFileSync("./src/app.rs", "utf8"), stream);
+stream.close();
+// Wigdet docs
+let data1 = fs.readFileSync("./src/widgets/widget_definitions.rs", "utf8");
+printDocs(parseVars(data1), parseDocs(data1));
+
+function parseMagicVariables(data, stream) {
+    const pattern = /^.*\/\/ @desc +(.*)$/;
+    for (let line of data.split("\n")) {
+        let match = line.match(pattern);
+        if (match) {
+            let split = match[1].split("-");
+            let name = split[0].trim();
+            let desc = split[1].trim();
+            stream.write(`### \`${name}\`\n${desc}\n`);
+        }
+    }
+}
```

Comment:
Not really a fan of the stream stuff here, as it seems,.... definitely unnecessarily complex for no real benefit.

Also, `printDocs` should be renamed to something more fitting if it now writes to the widgets file directly (which i'm also not really a fan of, as that means that the script is a lot more anoying to test now .-.)

Reply:
I'd argue in favour of just making a generateVarDoc function that does all of the variable documentation generation, and a generateWidgetDoc function (doing the same for the widget stuff), and then maybe just writing those to the respective files directly. Then i'd also be fine with writing directly to the files i guess

Reply:
Was thinking of doing that, but i didn't want to spend any more time in js, than necessary ;-;
My way of debugging has been generating the docs and then just `git restore docs/...` to the original state, without generated content. 
And i didn't really want to touch your js code, as i didn't really understand it or had the desire too -.-

Reply:
to be honest, just get rid of the streaming stuff (really not necessary) and maybe make the naming a bit more consistent - I've been planning to rework the doc script completely anyways (I want to use it to generate docs for everything, adding some more flexibility by allowing for a more complex widget structure - I really dislike having any specific documentation for xml-tags/properties not be synced with the code, so I'm going to try to have nearly _all_ of the docs be generated from comments in the code)